### DISCUSSION TITLE
Getting rid of "Commemoratio ad Laudes tantum"

### DISCUSSION THREAD

**Discussion by Augustinus-Altovadensis** [DivinumOfficium/divinum-officium#3414] [Q&A]
Laudetur Jesus Christus!

I'm trying to push forward the Cistercian version, and now I've realized that the normal Monastic version is very much "Romanized", and omits much of the Commemorations on Sundays and higher feasts. 

The Cistercian ordo has _first_ Vespers for all the Feasts, even the commemorated ones. Such Commemoration can only be omitted, if the main Feast is Sermo major (equivalent of 1st Class or Duplex major). Meanwhile _second_ Vespers are limited to Feasts with 12 Lessons.  

Now the current Monastic version has way too much Commemorations on Lauds only.

My question: is there an easy way, how to fix it, or do I have to read through the unbelievably complex Commemoration routine and try to make the heads or tails myself?

Thank you, God bless!

**Comment by FAJ-Munich** [comment]
… In æternum, Amen!

I cannot recommend to read through the Commemoration routine! First, it is unbelievably complex. Worse, it is utterly wrong: This is why every now and then, there is the necessity of some quick fixes like #3416 just recently. But, for instance, if you check out Vespers on Saturday 2023-10-28 under Divino Afflatu rules, you will notice the commemoration of the Sunday comes before the commemoration of Ss. Simon and Jude App. This is one of the instances where DA changed the order by directing the comm. of the concurrent office to come first always: which the algorithm is trying to accomplish but fails to do!

I am currently working on a new routine for Occurrence, Concurrence and Commemorations to replace the current function "getrank()" and ammend the "oratio" routine in specials.pl: hopefully, when this work is completed in a couple of days, this plethora of "quick fixes" comes to an end. After all, if we have a seperate database file for all the exceptions, one could simply get rid of the code and just hard code everything in the database files for each case seperately, which is to redicule the whole Div. Off. project, isn't it?

When it comes to the Cistercian ordo, I have to confess I have no idea on its rubrics. Your observation on the Monastic version, however, comes from the fact that 1963 is implemented which is mimicing the Roman 1960. Once we get to bring back Monastic 1617 and Monastic 1930, which is also on my personal agenda, there might be a better starting point to develop what you are looking for yourself.

**Reply by Augustinus-Altovadensis** [reply]
Oh gosh, I'm looking forward, maybe that will help me find the motivation to rebase, as that will take me quite a lot of time!

Yes, exactly. It would be ideal to write such a code that adding a new rite could comprise only in editing database files and some directives. But it's not that easy, I guess, and it's not how Laszlo wrote it in the beginning...

Actually, the Cistercian ordo, which I'm recreating, and that is being used in Vyšší Brod Abbey, is very much like the Monastic ordo before 1963. The brothers decided to return to 1951, the year the Monastery was closed down by the communists. It was made possible by Summorum Pontificum, that stated that the liturgical books from 1962 can still be used, as the Cistercian rite was reformed only in 1965 (and destroyed by novus ordo around 1970).

So yes, bringing back Monastic 1930 would help, although I'm still using a simplification by an advice I got here, that I call it "Cistercian Monastic", so unless stated otherwise, the code uses the Monastic part for everything. I would still need to change the basics, like ferial Hymns and most of the ferial Antiphons, the Commune Sanctorum is also different in small things like the Versicles, Antiphons for Magnificat used with Benedictus and other way around... So be it as it may, I might still be better off separating from the Monastic version altogether. The main issue however is time, it all of it requires a lot of it, although it might not be that hard now, when I know the code much better than in the beginning...

**Reply by Augustinus-Altovadensis** [reply]
So after some time and work, I have accomplished part of what I wanted, but it seems that not everything will be possible or at least reasonably feasible, so I'm (almost) ready to either update, or - which will probably be easier - start almost again, and implement the changes for Cistercian office into the code, that has remained, and figure out the Commemorations again. The data files should remain the same, shouldn't they? 

My main question is: I have seen a new `occurrence` routine in `horascommon.pl`. Is it the routine that you were talking about? And is it ready, or do you plan on some substantial updates quite soon? 

Thanks!

**Reply by FAJ-Munich** [reply]
@Augustinus-Altovadensis   yes, I was talking about the new `occurrence()` and `concurrence()` routine together with the changes at the end of the `oratio()` function of `specials.pl`. An important part of this routine is that I have "mathematically" encoded the various rubrics into the ranking of feasts and ferias which are explained in the new section of `technical.htm`
I hope I have added enough comments for outlining the logic behind the algorithm but please reach out if you have questions. Happy to explain my thoughts. Here is my Excel made overview of all the various occurrence rules between the versions (up to 1954):

![231210_Occurence](https://github.com/DivinumOfficium/divinum-officium/assets/60485740/722e75ff-4098-4eb3-9c68-20b2e4fbefac)

For unfinished work on the former Monastic versions (Benedictine Tridentine 1617) and (Benedictine Divino Afflatu 1930) you can have a look into the **monastic** branch in my form if you want. Maybe this can give you more ideas on what is possible or necessary to accomplish earlier versions; don't know how close the Cistercian is to the Benedictine.

**Reply by Augustinus-Altovadensis** [reply]
Very good, thank you! I have started the update, it will take a while to incorporate everything, and finding out, what even I have to change. The system of ranks in Cistercian calendar is somewhat different, as Cistercian never double antiphons, they have completely different names for the ranks (from the lowest: _Commemoratio, Commemoratio et Missa, Festum iij. Lectionum, Festum xij. Lectionum, Sunday, Festum MM. (duarum Missarum) minor, MM. maj., Festum sermonis minor, Serm. maj._). The rubrics are definitely before 1963, but it is not always identical to Tridentine rubrics... I think it might be similar to 1932 Monastic rubrics, but from what I read, most of it comes from 1888 and is basically just copied over from older sources, and there are some adjustments to Divino afflatu (ferial psalms on lower feasts iij. Lect. et M., basically).

But most of the texts are just different. Not by much, but it's like that Versicles that are in common texts for Vespers in Monastic rite, are in Lauds in the Cistercian rite, and some are just different. This I will need to take care about myself, it's not an issue. 

And I will see, whether the rank-dependent rubrics are going to be at least similar. 

For example, the rule for Commemorations is quite simple. Advent and Lent Ferias are commemorated always, Otherwise, everything is commemorated, except for most low rank feasts colliding with Serm. maj. on first Vespers. OK, we'll see :-)

**Reply by FAJ-Munich** [reply]
From what you described, I think there is good hope you can actually make it work relatively easily. 

Of course, the textual changes need to be incorporated into the database and the fact that Monastic folder only cares about 1963 makes this task more cumbersome. 

However, the naming of ranks is sort of irrelevant for how the code determines precedence and commemorations. It uses the rankings (with 2.1 (Feria major, I.e. Adv, Lent, QT) 2.99 (Dominica major at 2nd Vespers) 3.9 (Feria Quad 1960) and 4.2 and above always being commemorated) and sometimes the Latin names like Dominica, Feria, Octava, ... to determine the behaviour. 
So to achieve what you need you can add a Cisterician version of `setheadline()` to map the name which looks like the following:

- Commemoratio (encoded as rank `1.0`
- Festum iij. lect. (Sounds like Simplex encoded as `1.1`; not commemorated in Duplex I. cl. and, in II. cl., ad Laudes tantum)
- Festum xij. lect. (Sounds like Semiduplex; beware that Monastic 1930 has abolished this rank already except for Sundays and infra Octavam; octaves in Monastic never had 12 lessons in contrast to Roman which has had 9 lessons for these up to 1954) if you want the Roman behaviour of commemorations, encode it as 2.2; if you need commemorations at both vespers even in occurrence with serm. min., encode as 2.1 and take extra care of those occurring in lent and advent. (Maybe you have to invent a new ranking code, e.g. 2.11 for these specially and include this code in the `occurrence()` and `concurrence()` routine to be commemorated always)
- MM. minor Sounds Like Duplex ( encode as 3.0)
- MM. major Sounds Like Duplex major (encode as 4.0)
- Serm. minor ~= Duplex II. cl.
- Serm. major ~= Duplex I. cl. 

What you need to check if there is a discrepancy between Cistercian and Monastic 1930 (or Roman DA) on what rank will trigger Festal psalmody. For Monastic 1930, if I remember correctly, Duplex majus is festal and Duplex minus is still ferial. 

In summary, I think you can achieve what you need fairly easily when making use of the new ranking encoding. Once I am not on my mobile but desktop, I can also paste a screenshot of the logic behind the concurrence routine like I have done for the occurrence already. Stay tuned!

**Reply by FAJ-Munich** [reply]
Here's the promised screenshot:
![231216_Concurrence](https://github.com/DivinumOfficium/divinum-officium/assets/60485740/65454936-0d4b-4031-b59a-d68e5f42777a)

The more I think about your description of the Commemorations rule, and looking at the "pCn" entries in the row for 2nd Vespers of Duplex II. cl. with Semiduplex, I think you may be best of making use of the Tridentine rankings and rubrics but exchanging the Psalms from lower ranking feasts to the Ferial ones. 

> I'm still using a simplification by an advice I got here, that I call it "Cistercian Monastic"

As you said it above, you can try to exploit even more of the routines if you name it something like _Cistercian Monastic Tridentine_ or _Cistercian Monastic Divino_ depending on which set of rules bring you closer to what you need.

**Reply by Augustinus-Altovadensis** [reply]
Oh my, thanks a lot!

The logic is quite complicated, but understandable... and for what I can say, a bit different in the Cistercian rite. 
- in the Cistercian rite, there are no ranks that don't get commemorated. This is valid for the simple Commemoratio, as well as for anything else. This is also something I was fighting with in the previous version. The only exception is that certain _Serm. maj._ feasts don't have any commemoration on first Vespers. I mean such is the rule, but it gets overridden quite frequently in the Antiphonary.
- also, the Advent and Lent ferias get commemorated always, no matter the above rule.

But the solution might be quite simple - the lowest rank won't be 1.0, but 2.0 or something to get over that rule. Or if the rules are all in one place, not scattered all over `horascommon.pl` and `specials.pl` with occasional mentions elsewhere, I can just change it in the code.

I don't really know, whether I should even try to harmonise somehow the system of ranks with Monastic 1930, because it's quite different to anything else (same goes with the Classes 1 through 4 from 1961).

BTW, does the code in **horas.dialog** allow using something like `Ordo Cisterciensis - 1951/Cistercian Monastic Tridentine` (like in the case of language names), while showing only the first part, the `$version` variable would contain the entire text?

**Reply by FAJ-Munich** [reply]
> BTW, does the code in horas.dialog allow using something like `Ordo Cisterciensis - 1951/Cistercian Monastic Tridentine` (like in the case of language names), while showing only the first part, the `$version` variable would contain the entire text?

This one I don't know; haven't tried to hide the naming in my other branches.

> Or if the rules are all in one place, not scattered all over horascommon.pl and specials.pl with occasional mentions elsewhere, I can just change it in the code.

I think that would be the best solution such that you keep everything under control with future updates if they ever were to occur (which I don't hope for)

Within `horascommon.pl` the following parts take care of the Commemorations which you might need to change for Cistercian rubrics:

1. Code lines 243 – 263 are sorting out low ranking Sanctorals when there is a solemn feast in the Temporal and further changes for 1955 and 1960 that shouldn't interfere with your rubrics:
```
if (($trank[2] >= (($version =~ /19(?:55|6)/i) ? 6 : 7) && $srank[2] < 6)
	|| ($trank !~ /Dominica|Feria|Sabbato/i && (($trank[2] >= 6 && $srank[2] < 2.1)			# on Duplex I. cl nothing of Simplex and common octaves
	|| ($trank[2] >= 5 && $srank[2] == 2 && $srank[2] =~ /infra octavam/i)))					# on Duplex II. cl nothing of common octaves
	|| ($version =~ /19(?:55|6)/i && (($srank =~ /vigil/i && $sday !~ /(06\-23|06\-28|08\-09|08\-14|12\-24)/) || ($srank =~ /(infra octavam|in octava)/i && nooctnat()))) # TODO: to be made obsolte s.a.
	|| ($version =~ /1960/ && $dayofweek == 0 && (($trank[2] >= 6 && $srank[2] < 6) || ($trank[2] >= 5 && $srank[2] < 5)))) {
	$srank = '';
	%saint = undef;
	$sname = '';
	@srank = undef;
	@commemoentries = ();
} elsif ($version =~ /196/
	&& (($srank[2] >= 6 && $trank[2] < 6 && !($trank[2] == 2.1 || $trank[2] == 3.9 || $trank[2] == 4.9 || $trank[0] =~ /Dominica/i))
		|| ($trank[0] =~ /Dominica/i && $dayname[0] !~ /Nat1/i && $trank[2] <= 5 && $srank[2] >= 5 && $saint{Rule} =~ /Festum Domini/i))) {
	$tname = $trank = '';
	@trank = undef;
	%tempora = undef;
}	elsif ($version =~ /1955/ && $srank[2] >= 2 && $srank[2] <= 2.2 && $srank[1] =~ /Semiduplex/i) {
	$srank[2] = 1.5; #1955: semiduplex reduced to simplex
} elsif ($version =~ /196/i && $srank[2] < 2 && $srank[1] =~ /Simplex/i	&& $testmode =~ /seasonal/i	&& ($month > 1 || $day > 13))	{
	$srank[2] = 1;
}
```

2. When winner is Sanctoral, code lines 348–354 discards Temporal Feria minor (rank below 2) and Temporal Octava communi (S. Joseph) in case the winner is Duplex II. cl. or higher and keeps all the others
```
elsif(($srank[2] < 7 && $sname !~ /01-01/) && $trank[2] >= ($srank[2] >= 5 ? 2.1 : 2) && !($srank[0] =~ /Sangu/i && $trank[0] =~ /Cord/i)) { # incl. github #2950
	unshift(@commemoentries, $tname);
	$commemoratio = $tname;
	$comrank = $trank[2];
	$cvespera = $tvesp;
	$officename[2] = "Commemoratio: $trank[0]";
	$officename[2] =~ s/:/ ad Laudes tantum:/ if $tfile =~ /Pasc5\-[123]/i; # on Rogations and QT in Sep
}
```
3.  Please notice that the adding of `ad Laudes tantum` to the Commeoration string in lines 354 and 362–368 is just for cosmetics and has no influence on the code behaviour on its own.
4. When winner is Temporal, code lines 478–486 and 496–594 are just cosemetics again like mentioned before.
5. When it comes to sorting out Concurrence, I guess you really need to think through the various cases in the `concurrence()` routine. The logic goes as follows:

- First we sort out all the cases where there is no first Vespers of the next day (lines 629~672)
```
if($cwinner{Rule} =~ /No prima vespera/i
	|| ($version =~ /1955/ && $cwrank[2] < 5)	# Reduced 1955: No 1st Vespers except for Duplex I. cl & II. cl & Dominica
	|| ($version =~ /196/i && $cwrank[2] < (($cwrank[0] =~ /Dominica/i || ($cwinner{Rule} =~ /Festum Domini/i && $dayofweek == 6)) ? 5 : 6)) # In 1960, II. cl. feasts have I. Vespers if and only if they're feasts of the Lord on a Sunday.
	|| ($cwinner{Rank} =~ /Feria|Sabbato|Vigilia|Quat[t]*uor/i && $cwinner{Rank} !~ /in Vigilia Epi|in octava|infra octavam|Dominica|C10/i) # no Ferias, Vigils and infra Oct days
	|| ($cwinner{Rank} =~ /infra octavam/i && $cwinner{Rank} !~ /Dominica/i && ($version =~ /trident/i || $sanctoraloffice == $csanctoraloffice))  # before DA infra octavam always gets commemorated as at 2nd Vespers; after DA also when the office is of the octave
	|| ($weekname =~ /Pasc[07]/i && $cwinner{Rank} !~ /Dominica/i)	# infra 8vam Pasch & Pent
	|| ($winner =~ /01-01/ && $version !~ /trident/i)							# no commemoration of the Octave of S. Stephen after DA
	|| ($cwinner{Rank} =~ /C10/i && $winner{Rank} =~ /C1[01]/i)		# sort out BVM concurrent with BMV
	|| ($version =~ /19(?:55|6)/ && $cwinner{Rank} =~ /Dominica Resurrectionis|Patrocinii S. Joseph/i) # no 1st Vespers of Easter after 1955
	|| ($version =~ /19(?:55|6)/ && ($cwinner{Rank} =~ /octav/i && $cwinner{Rank} !~ /cum Octav/i && $cwrank[2] < 6))) {		# TODO: last condition should be made obsolete and handled via database
```
in some cases, we do keep the Commemoration of the Sanctoral (which is not the winner of the next day) (lines 639–649)
```
if ($ccomrank >= ($rank >= ($version =~ /trident/i ? 6 : 5) && $wrank[0] !~ /feria|octava/i ? 2.1 : 1.1) && $version !~ /1955|196/) {
	$vespera = 3;
	$dayname[2] = $tomorrowname[2] . "<br>Vespera de Officium occurente, Commemoratio Sanctorum tantum";
	$cwrank = '';
	$ctname = '';
	%cwinner = undef;
	@cwrank = undef;
	$cwinner = '';
	$crank = 0;
	$cvespera = 0;
}
```

- Next, we check the concurrence of two Tempora as these usually exclude each other from being concurrent (lines 673–702). In these cases, we do keep superseded sanctorals if these were not discarded previously as mentioned in point 1.
- Thereafter comes the general case in lines 703–819. The ìf-else` logics are pretty self explantory and literally saying what is happening in the lines where the Vespers descriptions are added to `$dayname[2]`. The most important part to look at for you are the first to possibilities where we discard commemrations of the Concurrent office, i.e. 
```
if (($rank >= (($version =~ /19(?:55|6)/) ? 6 : 7) && $crank < 6) # e.g. 05-26-2022
		|| ($version =~ /196/ && ($cwinner{Rank} =~ /Dominica/i && $dayname[0] !~ /Nat1/i && $crank <= 5)	&& ($rank >= 5 && $winner{Rule} =~ /Festum Domini/i)) #on a II. cl Sunday nothing at 1st Vespers in concurrence with a Feast of the Lord
		|| ($rank >= ($version =~ /trident/i ? 6 : 5) && $winner !~ /feria|in.*octava/i && $crank < 2.1)) {		# on Duplex I. cl / II. cl no commemoration of following Simplex and Common Octaves
	$dayname[2] .= "<br>Vespera de præcedenti; nihil de sequenti";
	$cwinner = '';
	%cwinner = ();
	$vespera = 3;
	$crank = 0;
	$cvespera = 0;
	@ccommemoentries = ();
	$ccomrank = 0;
} elsif ($rank < 2		# no 2nd Vespers of a Simplex
		|| ($version =~ /196/ && $cwinner{Rank} =~ /Dominica/i && $rank < 5) # on any Sunday, nothing of a preceding III. cl feast
		|| ($crank >= 6 && !($rank == 2.1 || $rank == 2.99 || $rank == 3.9 || $rank >= 4.2 ) && $cwrank[0] !~ /Dominica|feria|in.*octava/i)  # in 1st Vespers of Duplex I. cl. only commemoration of Feria major, Dominica (major), 8va privilegiata and Duplex II./I. cl
		|| ($cwinner =~ /12-25|01-01/) # on Christmas Eve and New Year's Eve, nothing of a preceding Sunday
		|| ($crank >= 5 && !($rank == 2.1 || $rank >= 2.99) && $cwrank[0] !~ /Dominica|feria|in.*octava/i)) { # in 1st Vespers of Duplex II. cl. also commemoration of any Duplex
	@dayname = @tomorrowname;
	$vespera = 1;
	$cvespera = 3;
	if ($comrank == 2.1 || $comrank == 2.99 || $comrank == 3.9) { # privilidged Feria, Dominica, or infra 8vam
		$dayname[2] .= "<br>Vespera de sequenti; commemoratio de off. priv. tantum";
	} else {
		$dayname[2] .= "<br>Vespera de sequenti; nihil de præcedenti";
	}
	$rank = $crank;
	$commune = $ccommune;
	$communetype = $ccommunetype;
	$winner = $cwinner;
	$cwinner = '';
	%cwinner = ();
}
```

6. After Concurrence is settled and it is clear whether we are in 1st Vespers of tomorrow or in 2nd Vespers of today, the algorithm further sorts out "unwanted" commemorations in lines 825–885, i.e., the last part of the `concurrence()` routine. Most of the sorting out logic is controled and encoded with the varios values for the `$ranklimit` variable. 
```
if ($vespera == 3) {	# We have 2nd Vespers:
	my $ranklimit = ($rank >= ($version =~ /trident/i ? 6 : 5) && $wrank[0] !~ /Dominica|feria|in.*octava/i) ? 2.1 : 1.1;
	# In Concurrence (i.e. tomorrow): Duplex I. cl (or also II. cl) and not "just" a privileged Feria or day (in/of) Octave exludes Simplex and infra 8vam communis
	my @comentries = ();
	my %cstr = ();
	foreach $commemo (@ccommemoentries) {
		if (!(-e "$datafolder/Latin/$commemo") && $commemo !~ /txt$/i) { $commemo =~ s/$/\.txt/; }
		%cstr = %{officestring('Latin', $commemo, 1)};
		if (($commemo =~ /tempora/i || $cstr{Rank} =~ /infra octavam/i) && $cstr{Rank} !~ /Dominica/i) { next; }	# no superseded Tempora or day within octave can have 1st vespers unless a Sunday
		if (%cstr) {
			my @cr = split(";;", $cstr{Rank});
			unless ($cr[2] < $ranklimit || $cstr{Rule} =~ /No prima vespera/i || $version =~ /1955|196/) { push(@comentries, $commemo); }
		}
	}
	@ccommemoentries = @comentries;
		
	# In Occurence (i.e. today): Simplex end after None.
	$ranklimit =  ($wrank[0] =~ /Dominica|feria|in.*octava/i) ? 2 : ($rank >= 6 && $version !~ /trident/i) ? 4.2 : $rank >= 5 ? 2.1 : 2;
	@comentries = ();
	foreach $commemo (@commemoentries) {
		if ($commemo =~ /tempora/i && $trank[2] < 2) { next; }	# Feria minor and Vigils have no Vespers if superseded
		if (!(-e "$datafolder/Latin/$commemo") && $commemo !~ /txt$/i) { $commemo =~ s/$/\.txt/; }
		%cstr = %{officestring('Latin', $commemo, 0)};
		if (%cstr) {
			my @cr = split(";;", $cstr{Rank});
			unless (($cr[2] < $ranklimit && !($cr[2] == 2.1 || $cr[2] == 2.99 || $cr[2] == 3.9)) || $cstr{Rule} =~ /No secunda vespera/i) { push(@comentries, $commemo); }   # sort out Simplex
		}
	}
	@commemoentries = @comentries;
} else { # We have 1st Vespers
	my $ranklimit = ($rank >= 6 && $wrank[0] !~ /Dominica|feria|in.*octava/i) ? 4.2 : ($rank >= 5 && $wrank[0] !~ /Dominica|feria|in.*octava/i) ? 2.99 : 2;
	# in Concurrence (i.e. today): Duplex I. cl excludes below Duplex II. cl., Duplex II. cl exludes below Duplex; Simplex have no 2nd Vespers
	my @comentries = ();
	my %cstr = ();
	@comentries = ();
	foreach $commemo (@commemoentries) {
		if ($commemo =~ /tempora/i && $trank[2] < 2) { next; }	# Feria minor and Vigils have no Vespers if superseded
		if (!(-e "$datafolder/Latin/$commemo") && $commemo !~ /txt$/i) { $commemo =~ s/$/\.txt/; }
		%cstr = %{officestring('Latin', $commemo, 0)};
		if (%cstr) {
			my @cr = split(";;", $cstr{Rank});
			unless (($cr[2] < $ranklimit && !($cr[2] == 2.1 || $cr[2] == 2.99 || $cr[2] == 3.9)) || $cstr{Rule} =~ /No secunda vespera/i || $cr[0] =~ /De VII di/i) { push(@comentries, $commemo); }   # sort out (Semi-)duplex and infra 8vam communis except for Feria major / Dominica major
		}
	}
	@commemoentries = @comentries;
		
	# In Occurencce (i.e. tomorrow):
	$ranklimit =  ($wrank[0] =~ /Dominica|feria|in.*octava/i) ? 1.1 : ($rank >= 6 && $version !~ /trident/i) ? 4.2 : $rank >= 5 ? 2.2 : 1.1;
	@comentries = ();
	foreach $commemo (@ccommemoentries) {
		if (!(-e "$datafolder/Latin/$commemo") && $commemo !~ /txt$/i) { $commemo =~ s/$/\.txt/; }
		%cstr = %{officestring('Latin', $commemo, 1)};
		if (($commemo =~ /tempora/i || $cstr{Rank} =~ /infra octavam/i) && $cstr{Rank} !~ /Dominica/i) { next; }	# no superseded Tempora or day within octave can have 1st vespers unless a Sunday
		if (%cstr) {
			my @cr = split(";;", $cstr{Rank});
			unless ($cr[2] < $ranklimit || $cstr{Rule} =~ /No prima vespera/i || ($version =~ /1955|196/ && $cstr{Rank} !~ /Dominica/i) || ($cstr{Rank} =~ /Feria|Sabbato|Vigilia|Quat[t]*uor/i && $cstr{Rank} !~ /in Vigilia Epi|in octava|Dominica/i)) { push(@comentries, $commemo); }
		}
	}
	@ccommemoentries = @comentries;
}
```


If you organise your database well, I think these parts of the routine should be sufficient to be adjusted to your needs. However, as I know that many database files are still plastered with random commemorations within the files itself, you may also have to have a look into `specials.pl`. All the commemorations are collected and printed at the end of the `oratio()`routine. 
7. Within Duplex I. cl, in-file commemorations are sorted out in lines 1284, 1365, and 1438 for the winner, cwinner, and commemo files.
```
unless(($rank >= 6 && $dayname[0] !~ /Pasc[07]|Pent01/) ...
```

8. The commemoration of Vigils in the office during Advent and Lent is suppressed in line 1643 of `vigilia_commemoratio()`
```
} elsif ($dayname[0] =~ /Adv|Quad[0-6]/i || ($dayname[0] =~ /Quadp3/i && $dayofweek >= 4)) {
	return '';
}
```

+++ Prope est iam Dóminus, * Veníte, adorémus. +++

**Reply by mbab** [reply]
> BTW, does the code in **horas.dialog** allow using something like `Ordo Cisterciensis - 1951/Cistercian Monastic Tridentine` (like in the case of language names), while showing only the first part, the `$version` variable would contain the entire text?

This will show `Ordo Cisterciensis - 1951` in options and set `$version = 'Cistercian Monastic Tridentine'`

**Reply by FAJ-Munich** [reply]
> This will show Ordo Cisterciensis - 1951 in options and set $version = 'Cistercian Monastic Tridentine'

@Augustinus-Altovadensis  I think you need to CherryPick the commit 434cf6d0bdb131ca53693472c30412347282e9e6 from my monastic branch to make it work. At least on my side it did not function immediately without that change.

**Comment by mbab** [comment]
I see Your repo is far behind so I heartily advise to keep it update, there where many changes this year.

**Reply by Augustinus-Altovadensis** [reply]
Yes, I hear you. I know I should do that, but with the changes I have made, I would have to go through the entire thing, which would take an amount of time I currently don't have... And as I use the app almost every day, I prefer having it working before having it up to date.

**Reply by mbab** [reply]
b968e88b2 & 322738fca separates Monastic from 1960
